### ISSUE TITLE
createRemoteDir doesn't always work : "The system cannot find the file specified."

### ISSUE THREAD

**Issue by ooooo84** [robinrodricks/FluentFTP#179]
I used this package to upload files to FTP server.
Using Client.Upload() method to upload file content with parameter createRemoteDir: true.
Sometimes it works well, most time it cannot work and throw "The system cannot find the file specified." exception.

Here is the success log:

```
\# Upload("/files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd/test-upload.txt", Overwrite, True)

\# FileExists("/files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd/test-upload.txt")

\# Connect()
Status:   Connecting to ***:21
Response: 220 Microsoft FTP Service
Command:  USER ***
Response: 331 Password required
Command:  PASS ***
Response: 230 User logged in.
Command:  FEAT
Response: 211 END
Response: 211-Extended features supported:
Response: LANG EN*
Response: UTF8
Response: AUTH TLS;TLS-C;SSL;TLS-P;
Response: PBSZ
Response: PROT C;P;
Response: CCC
Response: HOST
Response: SIZE
Response: MDTM
Response: REST STREAM
Status:   Text encoding: System.Text.UTF8Encoding
Command:  OPTS UTF8 ON
Response: 200 OPTS UTF8 command successful - UTF8 encoding now ON.
Command:  SYST
Response: 215 Windows_NT
Status:   Auto-detected Windows listing parser
Command:  SIZE /files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd/test-upload.txt
Response: 550 The system cannot find the path specified. 

\# DirectoryExists("/files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd")

\# GetWorkingDirectory()
Command:  PWD
Response: 257 "/" is current directory.
Command:  CWD /files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd
Response: 550 The system cannot find the file specified. 

\# CreateDirectory("/files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd", True)

\# DirectoryExists("/files/email-attachments/2017/09/07")

\# GetWorkingDirectory()
Command:  PWD
Response: 257 "/" is current directory.
Command:  CWD /files/email-attachments/2017/09/07
Response: 250 CWD command successful.
Command:  CWD /
Response: 250 CWD command successful.

\# DirectoryExists("/files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd")

\# GetWorkingDirectory()
Command:  PWD
Response: 257 "/" is current directory.
Command:  CWD /files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd
Response: 550 The system cannot find the file specified. 
Status:   CreateDirectory /files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd
Command:  MKD /files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd
Response: 257 "/files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd" directory created.

\# OpenWrite("/files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd/test-upload.txt", Binary)
Command:  TYPE I
Response: 200 Type set to I.

\# OpenPassiveDataStream(AutoPassive, "STOR /files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd/test-upload.txt", 0)
Command:  EPSV
Response: 229 Entering Extended Passive Mode (|||50234|)
Status:   Connecting to ***:50234
Command:  STOR /files/email-attachments/2017/09/07/93baddad-66e9-4023-a8a2-639aba890dbd/test-upload.txt
Response: 125 Data connection already open; Transfer starting.
Response: 226 Transfer complete.
Command:  QUIT
Response: 221 Goodbye.

\# Dispose()
Status:   Disposing FtpClient object...
Status:   Disposing FtpSocketStream...`


Here is the error log:
`
\# Upload("/files/email-attachments/2017/09/07/c6044c4c-984b-4ffe-bf89-5e792bca3351/test-upload.txt", Overwrite, True)

\# FileExists("/files/email-attachments/2017/09/07/c6044c4c-984b-4ffe-bf89-5e792bca3351/test-upload.txt")

\# Connect()
Status:   Connecting to ***:21
Response: 220 Microsoft FTP Service
Command:  USER ***
Response: 331 Password required
Command:  PASS ***
Response: 230 User logged in.
Command:  FEAT
Response: 211 END
Response: 211-Extended features supported:
Response: LANG EN*
Response: UTF8
Response: AUTH TLS;TLS-C;SSL;TLS-P;
Response: PBSZ
Response: PROT C;P;
Response: CCC
Response: HOST
Response: SIZE
Response: MDTM
Response: REST STREAM
Status:   Text encoding: System.Text.UTF8Encoding
Command:  OPTS UTF8 ON
Response: 200 OPTS UTF8 command successful - UTF8 encoding now ON.
Command:  SYST
Response: 215 Windows_NT
Status:   Auto-detected Windows listing parser
Command:  SIZE /files/email-attachments/2017/09/07/c6044c4c-984b-4ffe-bf89-5e792bca3351/test-upload.txt
Response: 550 End
Response: 550-The system cannot find the path specified.
Response: Win32 error:   The system cannot find the path specified.
Response: Error details: File system returned an error.
Command:  MDTM /files/email-attachments/2017/09/07/c6044c4c-984b-4ffe-bf89-5e792bca3351/test-upload.txt
Response: 550 End
Response: 550-The system cannot find the path specified.
Response: Win32 error:   The system cannot find the path specified.
Response: Error details: File system returned an error.

\# GetNameListing("/files/email-attachments/2017/09/07/c6044c4c-984b-4ffe-bf89-5e792bca3351")
Command:  TYPE I
Response: 200 Type set to I.

\# OpenPassiveDataStream(AutoPassive, "NLST /files/email-attachments/2017/09/07/c6044c4c-984b-4ffe-bf89-5e792bca3351", 0)
Command:  EPSV
Response: 229 Entering Extended Passive Mode (|||50338|)
Status:   Connecting to ***:50338
Command:  NLST /files/email-attachments/2017/09/07/c6044c4c-984b-4ffe-bf89-5e792bca3351
Response: 550 The system cannot find the file specified. 

\# CloseDataStream()
Command:  QUIT
Response: 221 Goodbye.

\# Dispose()
Status:   Disposing FtpClient object...
Status:   Disposing FtpSocketStream...
```

**Comment by hgupta9** [comment]
Can you try creating the directory yourself using CreateDirectory and see if that works for you? Then we'll be able to identify if that's the issue or it's something else..

**Comment by hgupta9** [comment]
And which is your server OS and FTP server?

**Comment by ooooo84** [comment]
I had made it works before by calling CreateDirectory method first before uploading files.
My FTP server is under IIS 8.0 and Windows Server 2012.

Hope it will help you to fix this issue.
Good job for the great component.

**Comment by robinrodricks** [comment]
Can you show the code you used that worked? Or at least the logs of the successful code? Are you calling CreateDirectory with slash at end?

**Comment by ooooo84** [comment]
The folder path is like this:
`Path.Combine(baseDir, Path.GetDirectoryName(path)).Replace('\\', '/')`

**Comment by daniel-nagy** [comment]
Same issue, explicitly creating the directory works. Windows Server 2012.

**Comment by olivierSOW** [comment]
Same issue when using createRemoteDir=true but works if i create directory before uploadling
here is the log:

> # Connect()
> Status:   Connecting to ::1:21
> Response: 220 Microsoft FTP Service
> Command:  USER anonymous
> Response: 331 Anonymous access allowed, send identity (e-mail name) as password.
> Command:  PASS ***
> Response: 230 User logged in.
> Command:  FEAT
> Response: 211 END
> Response: 211-Extended features supported:
> Response: LANG EN*
> Response: UTF8
> Response: AUTH TLS;TLS-C;SSL;TLS-P;
> Response: PBSZ
> Response: PROT C;P;
> Response: CCC
> Response: HOST
> Response: SIZE
> Response: MDTM
> Response: REST STREAM
> Status:   Text encoding: System.Text.UTF8Encoding
> Command:  OPTS UTF8 ON
> Response: 200 OPTS UTF8 command successful - UTF8 encoding now ON.
> Command:  SYST
> Response: 215 Windows_NT
> Status:   Auto-detected Windows listing parser
> 
> # UploadFile("D:\temp\favicon.ico - copie.png", "/test/test2/test.png", Overwrite, True, None)
> 
> # FileExists("/test/test2/test.png")
> Command:  SIZE /test/test2/test.png
> Response: 550 End
> Response: 550-
> Response: Win32 error:
> Response: Error details: File system returned an error.
> Command:  MDTM /test/test2/test.png
> Response: 550 End
> Response: 550-
> Response: Win32 error:
> Response: Error details: File system returned an error.
> 
> # GetNameListing("/test/test2")
> Command:  TYPE I
> Response: 200 Type set to I.
> 
> # OpenPassiveDataStream(AutoPassive, "NLST /test/test2", 0)
> Command:  EPSV
> Response: 229 Entering Extended Passive Mode (|||64319|)
> Status:   Connecting to ::1:64319
> Command:  NLST /test/test2
> Response: 550 End
> Response: 550-
> Response: Win32 error:
> Response: Error details: File system returned an error.
> Status:   Disposing FtpSocketStream...
> 
> # CloseDataStream()
> Exception levée : 'FluentFTP.FtpException' dans FluentFTP.dll

I work this dotnet2

**Comment by robinrodricks** [comment]
I tried to fix this in [FluentFTP 19.2.3](https://www.nuget.org/packages/FluentFTP/19.2.3). 

Please check if it works in your use case.
